xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp

```diff
@@ -59,6 +59,27 @@ bool CWinSystemWaylandEGLContextGL::InitWindowSystem()
   return true;
 }
 
+bool CWinSystemWaylandEGLContextGL::CreateSurface()
+{
+  const EGLint contextAttribs[] = {
+    EGL_CONTEXT_MAJOR_VERSION_KHR, 3,
+    EGL_CONTEXT_MINOR_VERSION_KHR, 2,
+    EGL_CONTEXT_OPENGL_PROFILE_MASK_KHR, EGL_CONTEXT_OPENGL_CORE_PROFILE_BIT_KHR,
+    EGL_NONE
+  };
+
+  if (!m_eglContext.CreateSurface(GetMainSurface(), GetBufferSize(), contextAttribs))
+  {
+    const EGLint fallbackContextAttribs[] = {
+      EGL_CONTEXT_CLIENT_VERSION, 2,
+      EGL_NONE
+    };
+    return m_eglContext.CreateSurface(GetMainSurface(), GetBufferSize(), fallbackContextAttribs);
```

Comment:
here we should log a warning that kodi won't show best peformance in legacy mode.